frontend/src/app/forms/forms.page.ts

```diff
@@ -172,4 +182,8 @@ export class FormsPage implements OnInit, OnDestroy {
       .length;
     this.progressPercentage = completedStep / this.totalSteps;
   }
+
+  getButtonText(): string {
```

Comment:
Don't bind  method to template unless and until you cannot achieve it with property binding. Just put a console.log() inside that method and check how many times it is called . Whenever you type or change menu in any step, this method will be called because of change detection mechanism in angular.